presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/BenchmarkCachingHiveMetastore.java

```diff
@@ -0,0 +1,517 @@
+            }
+
+            String[] partitionNames = new String[partitionsPerTableCount];
+            for (int i = 0; i < partitionNames.length; i++) {
+                ArrayList<String> values = new ArrayList<>(partitionColumns.length);
+                for (int j = 0; j < partitionColumns.length; j++) {
+                    values.add(randomAlphaNumericString(5));
+                }
+                partitionNames[i] = FileUtils.makePartName(Arrays.asList(partitionColumns), values);
+            }
+            cyclePartitionNames = cycle(partitionNames);
+        }
+
+        public HiveIdentity nextIdentity()
+        {
+            return new HiveIdentity(cycleIdentities.next());
+        }
+
+        public String nextDatabaseName()
+        {
+            return new String(cycleDatabaseNames.next());
+        }
+
+        public String nextTableName()
+        {
+            return new String(cycleTableNames.next());
+        }
+
+        public Set<String> nextPartitionsToSet(int size)
+        {
+            HashSet<String> partitions = new HashSet<>(size);
+            for (int i = 0; i < size; i++) {
+                partitions.add(new String(cyclePartitionNames.next()));
+            }
+            return partitions;
+        }
+
+        public List<String> nextPartitionsToList(int size)
+        {
+            ArrayList<String> partitions = new ArrayList<>(size);
+            for (int i = 0; i < size; i++) {
+                partitions.add(new String(cyclePartitionNames.next()));
+            }
+            return partitions;
+        }
+    }
+
+    private static String randomAlphaString(int size)
+    {
+        StringBuilder builder = new StringBuilder(size);
+        for (int i = 0; i < size; i++) {
+            builder.append(randomAlphaChar());
+        }
+        return builder.toString();
+    }
+
+    private static char randomAlphaChar()
+    {
+        ThreadLocalRandom rand = ThreadLocalRandom.current();
+        char base = rand.nextBoolean() ? 'a' : 'A';
+        return (char) (base + rand.nextInt(26));
+    }
+
+    private static String randomAlphaNumericString(int size)
+    {
+        StringBuilder builder = new StringBuilder(size);
+        builder.append(randomAlphaChar());
+        ThreadLocalRandom rand = ThreadLocalRandom.current();
+        for (int i = 1; i < size; i++) {
+            switch (rand.nextInt(3)) {
+                case 0:
+                    builder.append((char) ('a' + rand.nextInt(26)));
+                    break;
+                case 1:
+                    builder.append((char) ('A' + rand.nextInt(26)));
+                    break;
+                case 2:
+                    builder.append((char) ('0' + rand.nextInt(10)));
+                    break;
+            }
+        }
+        return builder.toString();
+    }
+
+    private static <T> CycleOfValues<T> cycle(T[] values)
+    {
+        return new CycleOfValues<T>(values);
```

Comment:
BTW I think this can just be `Iterables.cycle(values).iterator()`... there might be a way with stream also

Reply:
Yeah, I started with that but after looking at the implementation I chose to do this. I wanted to avoid adding any unnecessary overhead to that part of the benchmark.

Reply:
Yeah, I started there but the implementation didn't look very efficient. I was trying to avoid any unnecessary overhead to that part of the benchmark so just rolled my own simple one.